### PR TITLE
Feat implement translate text in dart

### PR DESCRIPTION
**Pull request by Rishabh-Negi**
Related Issue #97

Created translateText folder in dart directory with main.dart, pubspec.yaml and README.md.

Please review the PR and suggest edits. Thanks!

### CODE REVIEW

File: dart/translate_text/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+* **GOOGLE_APPLICATION_CREDENTIALS** - Your google json API key path
+* **PROJECT_ID** - Your google cloud project id
+
+## 🚀 Building and Packaging
+
+To package this example as a cloud function, follow these steps.
+
+```bash
+$ cd demos-for-functions/dart/translate_text
```
Comment (lohanidamodar) [inline dart/translate_text/README.md:15]: You must add a step to set PUB_HOME variable, check other dart function demos.

File: dart/translate_text/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+$ dart pub get
+```
+
+* Ensure that your folder structure looks like this 
+```
```
Comment (lohanidamodar) [inline dart/translate_text/README.md:20]: Folder structure must contain `.appwrite` folder with pub packages.

File: dart/translate_text/main.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  final parent = "projects/${envVars['PROJECT_ID']}";
+
+  final TranslateTextResponse result =
+      await translateApi.projects.translateText(request, parent);
+
+  return result.translations?[0].translatedText ?? "";
+}
+
+void main(List<String> args) async {
+  final text = 'Hello world';
```
Comment (lohanidamodar) [inline dart/translate_text/main.dart:38]: The text for translation should be received from the function execution payload.

File: dart/translate_text/main.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  final sourceLanguage = 'en-US';
+  final destinationLanguage = 'sr-Latn';
+
```
Comment (lohanidamodar) [inline dart/translate_text/main.dart:41]: may be also a validation to make sure required env vars are supplied?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]

File: dart/translate_text/main.dart (hunk 3 above)
Comment (Rishabh-Negi) [reply dart/translate_text/main.dart]: should the source language and destination language also be received from the function execution payload?
Thanks

File: dart/translate_text/main.dart (hunk 4 above)
Comment (Rishabh-Negi) [reply dart/translate_text/main.dart:41]: I added an If else block if some other type of validation is required please suggest.
Thanks

File: dart/translate_text/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+$ export PUB_CACHE=.appwrite/
+$ dart pub get
+```
+
+* Ensure that your folder structure looks like this 
+```
+.
+├── main.dart
+├── .appwrite
```
Comment (lohanidamodar) [inline dart/translate_text/README.md:25]: to make sure they know it's a directory not file
```suggestion
├── .appwrite/
```

File: dart/translate_text/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Translate text from one language to another
+A sample Dart Cloud Function for translating text from one language to another using google translate API.
+
+## 📝 Environment Variables
+Go to Settings tab of your Cloud Function. Add the following environment variables.
+
+* **GOOGLE_APPLICATION_CREDENTIALS** - Your google json API key path
+* **PROJECT_ID** - Your google cloud project id
+
```
Comment (lohanidamodar) [inline dart/translate_text/README.md:9]: As the function requires specific inputs in specific format, please include input/output section with proper details.

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
@Rishabh-Negi awesome work 👍🏻 We need some more changes to make it even better.

**Comment by lohanidamodar** [comment]
@Rishabh-Negi please also add screenshot of successful execution and output in the description

**Comment by christyjacob4** [comment]
@Rishabh-Negi please add screenshots of successful executions. You can take a look at some of the merged PRs for reference.

**Comment by christyjacob4** [comment]
@Rishabh-Negi closing this PR for now since there has been no activity in quite a long time. Feel free to reopen it when you're ready to start working on it once again. 🙂

**Pull request closed without merging**